swan-lake/learn-the-platform/run-in-the-cloud/function-as-a-service/azure-functions.md

```diff
@@ -1,379 +1,148 @@
-3. Once the database is created, go to the **Data Explorer**, and select **Create Container**.
-4. Enter `db1` as Database ID and `c1` as the collection ID, and click **Ok**.
+## Build the function
 
-**Note:** If you want to change these values, make sure to change them in the code as well.
-5. Go to the **Keys** tab of the Cosmos DB page.
-6. Copy the value of the `PRIMARY CONNECTION STRING`.
-7. Click the **Configuration** tab on the Function app page.
-8. Select **New Application Setting**, and paste the data you copied above as the value. For the key, use the value of the `connectionStringSetting` key and save.
+The Azure Functions functionality is implemented as a compiler extension. Therefore, artifact generation happens automatically when you build a Ballerina module by executing the command below. 
 
-Example application setting is as follows.
 ```
-Name - `CosmosDBConnection`
-Value - `AccountEndpoint=https://db-cosmos.documents.azure.com:443/;AccountKey=12345asda;`
+$ bal build --cloud="azure_functions"
 ```
 
-Now, as all the infrastructure required are up and running and configured, start building and deploying the Azure function.
-
-```ballerina
-import ballerina/log;
-import ballerinax/azure_functions as af;
-
-public type DBEntry record {
-    string id;
-    string name;
-};
+## Deploy the function
 
-@af:CosmosDBTrigger {connectionStringSetting: "CosmosDBConnection", databaseName: "db1", collectionName: "c1"}
-listener af:CosmosDBListener cosmosEp = new ();
+To deploy the function, execute the command, which you get in the CLI output logs after you [build the function](#build-the-function). For examples, see [Examples](#examples).
 
-service "cosmos" on cosmosEp {
-    remote function onUpdate(DBEntry[] entries) returns @af:QueueOutput {queueName: "people"} string {
-        string name = entries[0].name;
-        log:printInfo(entries.toJsonString());
-        return "Hello, " + name;
-    }
-}
+>**Note:** When you are deploying, make sure to replace the `<function_app_name>` placeholder with the app name of the [created function](#create-the-function).
 
-```
 
-Execute the `bal build --cloud="azure_functions"` command on the package directory to build the package. Then, execute the `func azure functionapp publish <function_app_name> --script-root target/azure_functions ` command shown in the Ballerina build output to deploy it.
+## Azure Functions native (experimental)
 
-Once the function is deployed, add an item to the collection.
-1. Navigate to the collection created in the **Data Explorer**.
-2. Click **New Item** to add a new item to the collection.
-3. Go to the queue page and observe the added new entry.
+You can use Azure Functions in the [native approach of Ballerina](/learn/build-the-executable-in-a-container/).
 
-**Info:** Additionally, for debugging purposes, view the logs under the **Logs stream** in the function app.
+### Set up the prerequisites for the native approach
```

Comment:
Do we need to repeat `native approach` part in each subtitle?

Reply:
+1 to remove

Reply:
Removed.